projects/ad7091r8-sdz/src/examples/iio_timer_trigger_example/iio_timer_trigger_example.c

```diff
@@ -0,0 +1,151 @@
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+/***************************************************************************//**
+ * @brief IIO trigger example main execution.
+ *
+ * @return ret - Result of the example execution. If working correctly, will
+ *               execute continuously function iio_app_run and will not return.
+*******************************************************************************/
+
+int iio_timer_trigger_example_main()
+{
+	struct ad7091r8_iio_dev *ad7091r8_iio_desc;
+	struct ad7091r8_iio_dev_init_param ad7091r8_iio_param;
+	struct iio_app_desc *app;
+	struct iio_app_init_param app_init_param = { 0 };
+	struct ad7091r8_init_param *ad7091r8_init_p;
+	struct iio_data_buffer adc_buff = {
+		.buff = (void *)ad7091r8_data_buffer,
+		.size = IIO_DATA_BUFFER_SIZE
+	};
+	int ret;
+
+	/* adc trigger instance descriptor. */
+	struct iio_hw_trig *ad7091r8_timer_trig_desc;
+
+	/* adc timer descriptor */
+	struct no_os_timer_desc *ad7091r8_timer_desc;
+
+	/* ad7091r8 irq instance descriptor */
+	struct no_os_irq_ctrl_desc *ad7091r8_irq_desc;
+
+
+	/* Use ad7091r4_ip and ad7091r2_ip for AD7091R-4 and AD7091R-2 respectively */
+	ad7091r8_init_p = &ad7091r8_ip;
+	ad7091r8_iio_param.ad7091r8_dev_init = ad7091r8_init_p;
+	ret = ad7091r8_iio_init(&ad7091r8_iio_desc, &ad7091r8_iio_param);
+	if (ret)
+		return ret;
+
+	ret = no_os_timer_init(&ad7091r8_timer_desc, &ad7091r8_timer_ip);
+	if (ret)
+		return ret;
+
+
+	ret = no_os_irq_ctrl_init(&ad7091r8_irq_desc, &ad7091r8_timer_irq_ip);
+	if (ret)
+		return ret;
+
+
+	ret = no_os_irq_set_priority(ad7091r8_irq_desc, AD7091R8_TIMER_IRQ_ID, 1);
+	if (ret)
+		return ret;
+
+
+	ad7091r8_timer_irq_ip.extra = ad7091r8_timer_desc->extra;
+
+	ad7091r8_timer_trig_ip.irq_ctrl = ad7091r8_irq_desc;
+
+	/* Initialize hardware trigger */
+	ret = iio_hw_trig_init(&ad7091r8_timer_trig_desc, &ad7091r8_timer_trig_ip);
+	if (ret)
+		return ret;
+
+	ret = no_os_timer_start(ad7091r8_timer_desc);
+	if (ret)
+		return ret;
+
+	struct iio_app_device iio_devices[] = {
+		IIO_APP_DEVICE(ad7091r8_names[ad7091r8_init_p->device_id],
+			       ad7091r8_iio_desc, ad7091r8_iio_desc->iio_dev,
+			       &adc_buff, NULL, NULL),
+	};
+
+	struct iio_trigger_init trigs[] = {
+		IIO_APP_TRIGGER(AD7091R8_TIMER_TRIG_NAME, ad7091r8_timer_trig_desc,
+				&ad7091r8_iio_timer_trig_desc),
+	};
+
+	/* Set device trigger */
+	iio_devices[0].default_trigger_id = "trigger0";
+
+	app_init_param.devices = iio_devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(iio_devices);
+	app_init_param.uart_init_params = ad7091r8_uart_ip;
+	app_init_param.trigs = trigs;
+	app_init_param.nb_trigs = NO_OS_ARRAY_SIZE(trigs);
+	app_init_param.irq_desc = NULL;
+
+	ret = iio_app_init(&app, app_init_param);
+	if (ret)
+		return ret;
```

Comment:
please fix the memory leaks in this function in case of an error...make sure you call all remove APIs if the init API has been successful